go/enclave/gas/gas.go

```diff
@@ -0,0 +1,54 @@
+package gas
+
+import (
+	"math/big"
+
+	gethcore "github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type ObscuroGasPool struct {
+	gp gethcore.GasPool
+}
+
+func NewObscuroGasPool(gp *gethcore.GasPool) *ObscuroGasPool {
+	return &ObscuroGasPool{
+		gp: gethcore.GasPool(gp.Gas()),
+	}
+}
+
+func (gp *ObscuroGasPool) ForTransaction(tx *types.Transaction) (*gethcore.GasPool, error) {
+	encodedTx, err := rlp.EncodeToBytes(*tx)
+	if err != nil {
+		return nil, err
+	}
+
+	l1Gas := CalculateL1GasUsed(encodedTx, big.NewInt(0))
+
+	gPool := gethcore.GasPool(l1Gas.Uint64())
+	return &gPool, nil
+}
+
+func CalculateL1GasUsed(data []byte, overhead *big.Int) *big.Int {
```

Comment:
this needs a comment.
If I were to guess, this tries to estimate how much it will cost storing this tx to the l1.
If yes, I don't think it's working correctly.
Because the transaction will be compressed and encrypted.
I would say, it can only work by estimating. Taking the size, reducing it to 66% and then multiply by the cost of a non-zero byte

Reply:
This is how optimism does it. I'm not sure if taking the size directly and assuming it is all 1s is the correct approach to go, cause zero to ones ratio should somewhat be maintained after compression, right?